### ISSUE TITLE
Is it feasible to qualify imports from Prelude?

### ISSUE THREAD

**Issue by JustusAdam** [haskell/alex#127]
## Problem

I happen to use Alex in a context where I have a custom `Prelude` and also
warnings enabled for the use of `undefined` **and** to top things off I use
`--pedantic` when compiling on CI.

So basically its a bit inconvenient.

My question therefore is would it not be better perhaps to only use fully
qualified imports and fully qualified names in the templates/lexers to avoid
potential clashes with user imports/defines?

## Solution

Because this is a very minor issue I'd happily take it upon myself to
"implement" it, so long as there are no objections.

**Comment by simonmar** [comment]
Yes, I think this would be good. I'll gladly accept a PR that fully qualifies everything.

**Comment by andreasabel** [comment]
@JustusAdam Would you still be up for this?
I added qualification for all non-Prelude imports in:
- #259

**Comment by andreasabel** [comment]
I suppose there is no more interest in this issue.
